### ISSUE TITLE
Macro sigils with uppercase letters

### ISSUE THREAD

**Issue by lasseebert** [rrrene/credo#682]
### Environment

* Credo version (`mix credo -v`): 1.1.2
* Erlang/Elixir version (`elixir -v`): Elixir 1.8.2 (compiled with Erlang/OTP 21)
* Operating system: Ubuntu

### What were you trying to do?

```elixir
defmacro sigil_U({:<<>>, _, [string]}, []) do
  # ...
end
```

### Expected outcome

No warnings

### Actual outcome

```
┃ [R] ↗ Function/macro/guard names should be written in snake_case.
┃       test/support/timing.ex:36:12 #(Balrog.Test.Timing.sigil_U)
```

This is related to https://github.com/rrrene/credo/issues/679, only difference is that this is a `defmacro` instead of a `def`.

**Comment by rrrene** [comment]
@lasseebert I lost track of this. Is this still an issue in Credo `v1.1.4`?

**Comment by lasseebert** [comment]
No it's not a problem anymore :)

I didn't really noticed, since I only used `sigil_U` for UTC datetimes in tests, but that was added natively in Elixir 1.9, so I removed my own definition.

I just tested on Credo 1.1.3 and 1.1.4 and it is no longer an issue :+1:
